Element84/filmdrop-ui
# Ba/feature/dynamic layer legend

**Related Issue(s):**

- NA

**Proposed Changes:**

1. Add Legend component for all layers
2. Consolidate hex legend into single layer legend
3. Change highlight color for better contrast with future feature (WIP) state

### To Test:

automated tests pass 🍏 

**Test basic legend**
- set values in config.json to:  `ADVANCED_SEARCH_ENABLED: false` & `CART_ENABLED: false`
- load app
- confirm no legend shows on load
- leave default settings
- hit search
- confirm no legend shows
- zoom in to grid-code aggregation level
- hit search
- confirm legend shows
- confirm only item showing in legend is 'Scene aggregation'
- confirm symbology matches map 
- zoom in to scene level
- hit search again
- confirm only item showing in legend is 'Available scene'
- change collection to be Cop DEM
- zoom out to hex level
- hit search
- confirm 'Aggregate Scenes' label and ramp is shown in legend

**Test cart legend**
- stop app
- set values in config.json to: `CART_ENABLED: true`
- reload app
- zoom into where scene mode is used
- click search
- confirm 'Scene in cart' and 'Available scene' both show in legend
- zoom out to grid-code level
- click search again
- confirm 'Scene in cart' no longer shows in legend
- change collection to be Cop DEM
- zoom out to hex level
- hit search
- confirm 'Scene in cart' does not show in legend

**Test draw legend**
- stop app
- set values in config.json to:  `ADVANCED_SEARCH_ENABLED: true`
- reload app
- click advanced search button
- draw boundary for search or upload geojson
- confirm legend shows
- confirm only 'Search area' shows in legend
- confirm symbology matches map
- click advanced search button
- click clear
- confirm legend no longer shows
- draw or upload geom again
- zoom into grid code level
- click search
- confirm both 'Search area' and 'Scene aggregation' legend items show
- zoom into single scene level
- click search
- confirm 'Search area', 'Available scene' and 'Scene in cart' legend items show
- zoom out to hex level
- change collection to be Cop DEM
- hit search
- confirm only 'Search area' and 'Aggregate Scenes' with color ramp show in legend items

**PR Checklist:**
